File: src/main/resources/static/js/emailconfig.js
document.addEventListener('DOMContentLoaded', function () {
    const form = document.getElementById('emailForm');
    const engineEl = document.getElementById('emailEngine');
    const smtpBlock = document.getElementById('smtpFields');
    const sesBlock = document.getElementById('sesFields');
    const hintEl = document.getElementById('smtpPasswordHint');
    const secretHintEl = document.getElementById('awsSecretHint');

    function setValue(id, value) {
        const field = document.getElementById(id);
        if (field) field.value = value == null ? '' : String(value);
    }

    function setHint(el, value) {
        if (!el) return;
        if (value) {
            el.textContent = value;
            el.hidden = false;
        } else {
            el.textContent = '';
            el.hidden = true;
        }
    }

    function normalizeEngine(value) {
        const engine = String(value || 'smtp').toLowerCase().replace(/\s+/g, '_');
        if (engine === 'sendmail') return 'sendmail';
        if (engine === 'aws_ses' || engine === 'awsses' || engine === 'ses') return 'aws_ses';
        return 'smtp';
    }

    function toggleEngine() {
        const engine = engineEl.value;
        smtpBlock.hidden = engine !== 'smtp';
        sesBlock.hidden = engine !== 'aws_ses';
    }

    function applyConfig(data) {
        engineEl.value = normalizeEngine(data.emailEngine);
        setValue('fromEmail', data.fromEmail);
        setValue('smtpUsername', data.smtpUsername);
        setValue('smtpPassword', '');
        setValue('smtpServer', data.smtpServer);
        setValue('smtpPort', data.smtpPort || '587');
        setValue('smtpSecurity', (data.smtpSecurity || 'tls').toLowerCase() === 'off' ? 'off' : (data.smtpSecurity || 'tls').toLowerCase());
        setValue('smtpAuth', (data.smtpAuth || 'on').toLowerCase() === 'off' ? 'off' : 'on');
        setValue('awsAccessKeyId', data.awsAccessKeyId);
        setValue('awsSecretAccessKey', '');
        setValue('awsRegion', data.awsRegion || 'us-east-1');
        setHint(hintEl, data.passwordHint);
        setHint(secretHintEl, data.awsSecretHint);
        toggleEngine();
    }

    engineEl.addEventListener('change', toggleEngine);

    form.addEventListener('submit', async function (event) {
        event.preventDefault();
        const payload = {
            emailEngine: engineEl.value,
            fromEmail: document.getElementById('fromEmail').value.trim(),
            smtpUsername: document.getElementById('smtpUsername').value.trim(),
            smtpPassword: document.getElementById('smtpPassword').value,
            smtpServer: document.getElementById('smtpServer').value.trim(),
            smtpPort: document.getElementById('smtpPort').value.trim(),
            smtpSecurity: document.getElementById('smtpSecurity').value,
            smtpAuth: document.getElementById('smtpAuth').value,
            awsAccessKeyId: document.getElementById('awsAccessKeyId').value.trim(),
            awsSecretAccessKey: document.getElementById('awsSecretAccessKey').value,
            awsRegion: document.getElementById('awsRegion').value
        };
        try {
            const response = await fetch('/api/emailconfig', {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify(payload)
            });
            const result = await response.json();
            if (!response.ok || result.success === false) throw new Error(result.message || 'Save failed');
            if (result.data) applyConfig(result.data);
            Swal.fire({ icon: 'success', title: 'Saved!', text: result.message, timer: 1400, showConfirmButton: false });
        } catch (error) {
            Swal.fire({ icon: 'error', title: 'Error', text: error.message, confirmButtonColor: '#8b5cf6' });
        }
    });

    fetch('/api/emailconfig')
        .then(function (response) { return response.json(); })
        .then(applyConfig)
        .catch(function (error) {
            toggleEngine();
            Swal.fire({ icon: 'error', title: 'Error', text: error.message || 'Failed to load settings', confirmButtonColor: '#8b5cf6' });
        });
});

File: src/main/java/com/kantechsolution/smart_school/model/EmailConfig.java
package com.kantechsolution.smart_school.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Lob;
import jakarta.persistence.Table;
import lombok.*;

@Entity
@Table(name = "email_config")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EmailConfig extends BaseEntity {

    @Column(name = "email_engine", nullable = false, length = 30)
    private String emailEngine;

    @Column(name = "from_email", length = 150)
    private String fromEmail;

    @Column(name = "smtp_username", length = 150)
    private String smtpUsername;

    @Lob
    @Column(name = "smtp_password", columnDefinition = "TEXT")
    private String smtpPassword;

    @Column(name = "smtp_server", length = 150)
    private String smtpServer;

    @Column(name = "smtp_port", length = 10)
    private String smtpPort;

    @Column(name = "smtp_security", length = 20)
    private String smtpSecurity;

    @Column(name = "smtp_auth", length = 10)
    private String smtpAuth;

    @Column(name = "aws_access_key_id", length = 100)
    private String awsAccessKeyId;

    @Lob
    @Column(name = "aws_secret_access_key", columnDefinition = "TEXT")
    private String awsSecretAccessKey;

    @Column(name = "aws_region", length = 40)
    private String awsRegion;
}
